rwengine/src/script/ScriptTypes.hpp

```diff
@@ -26,6 +29,145 @@ typedef uint16_t SCMOpcode;
 typedef char SCMByte;
 typedef unsigned int SCMAddress;
 
+///////////////////////////////////////////////////////////
+// Script Implementation Types
+
+using ScriptInt = int32_t;
+using ScriptLabel = int32_t;
+using ScriptIntGlobal = int32_t&;
+using ScriptIntLocal = int32_t&;
+using ScriptFloat = float;
+using ScriptFloatGlobal = float&;
+using ScriptFloatLocal = float&;
+using ScriptBoolean = int32_t;
+
+using ScriptVec2 = glm::vec2;
+using ScriptVec3 = glm::vec3;
+
+using ScriptRGB = glm::u8vec3;
+using ScriptRGBA = glm::u8vec4;
```

Comment:
Shouldn't a script RGB still consist of 3 script integers?
The bound checking + conversion might be handled elsewhere, I was just wondering.